### PR TITLE
N21-654 implementation of migration final finish

### PR DESCRIPTION
**Pull request by IgorCapCoder**
# Short Description
with the introduction of the oauthMigrationFinalFinish attribute, some logic changes and text adjustments are being made here

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-654
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4001
https://github.com/hpi-schul-cloud/dof_app_deploy/pull/492

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Review by arnegns — CHANGES_REQUESTED** [review]
Please add tests

File: src/store/env-config.ts [hunk 1]
```diff
@@ -94,6 +95,10 @@ export default class EnvConfigModule extends VuexModule {
 		return this.env.FEATURE_SCHOOL_SANIS_USER_MIGRATION_ENABLED;
 	}
 
+	get getMigrationEndGracePeriod() {
```
Comment (arnegns) [inline src/store/env-config.ts:98]: missing test

**Review by arnegns — APPROVED** [review]
Please add the test

File: src/store/env-config.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export const requiredVars = {
 export const configsFromEnvironmentVars = {
 	FEATURE_LERNSTORE_ENABLED:
 		process.env.FEATURE_LERNSTORE_ENABLED?.toLowerCase() == "true",
+	MIGRATION_END_GRACE_PERIOD: process.env.MIGRATION_END_GRACE_PERIOD,
```
[the remark this answers is not in the document]
Comment (IgorCapCoder) [reply src/store/env-config.ts]: ... and the default value is derived process.env. If you would like to see it in another way, i'm open to suggestions

File: src/components/administration/MigrationWarningCard.vue [hunk 3]
```diff
@@ -87,6 +100,16 @@ export default defineComponent({
 			eventName = "end";
 		}
 
+		const gracePeriodInDays: ComputedRef<number | undefined> = computed(() => {
+			const days: number | undefined = undefined;
+			if (envConfigModule?.getMigrationEndGracePeriod) {
+				const days = envConfigModule.getMigrationEndGracePeriod / 86400000;
```
Comment (arnegns) [inline src/components/administration/MigrationWarningCard.vue:106]: Please magic numbers in extra const

File: src/components/administration/MigrationWarningCard.vue [hunk 4]
```diff
@@ -87,6 +100,16 @@ export default defineComponent({
...
+				return days;
+			} else {
```
Comment (arnegns) [inline src/components/administration/MigrationWarningCard.vue:108]: You dont need a else if you return

**Review by arnegns — APPROVED** [review]

File: src/components/administration/AdminMigrationSection.unit.ts [hunk 5]
```diff
@@ -15,17 +17,21 @@ describe("AdminMigrationSection", () => {
 				oauthMigrationPossible: false,
 				oauthMigrationMandatory: false,
 				oauthMigrationFinished: "",
+				oauthMigrationFinalFinish: "",
```
Comment (MartinSchuhmacher) [inline src/components/administration/AdminMigrationSection.unit.ts:20]: whats the difference between "finished"  and "finalfinish"? word wise this is confusing, maybe changing variable names would improve readability

File: src/components/administration/AdminMigrationSection.unit.ts [hunk 6]
```diff
@@ -458,20 +503,72 @@ describe("AdminMigrationSection", () => {
 				oauthMigrationPossible: true,
 				oauthMigrationMandatory: false,
 				oauthMigrationFinished: "",
+				oauthMigrationFinalFinish: "",
 				enableMigrationStart: true,
 			});
 		});
 	});
 
+	describe("when checkbox is not clicked", () => {
```
Comment (MartinSchuhmacher) [inline src/components/administration/AdminMigrationSection.unit.ts:512]: clicked -> checked/unchecked? not clear whats the result of that click supposed to be

File: src/components/administration/AdminMigrationSection.unit.ts [hunk 7]
```diff
@@ -458,20 +503,72 @@ describe("AdminMigrationSection", () => {
...
+			await buttonComponent.vm.$emit("click");
+
+			const cardComponent = wrapper.findComponent({ name: "v-card" });
+			const cardButtonAgree = cardComponent.find("[data-testid=agree-btn]");
+
+			expect(cardButtonAgree.props("disabled")).toBeTruthy();
+		});
+	});
+
+	describe("when checkbox is clicked", () => {
```
Comment (MartinSchuhmacher) [inline src/components/administration/AdminMigrationSection.unit.ts:533]: same here

File: src/components/administration/AdminMigrationSection.unit.ts [hunk 8]
```diff
@@ -483,13 +580,37 @@ describe("AdminMigrationSection", () => {
 			);
 		});
 
+		it("should show finalFinish text when migration grace period has expired", async () => {
```
Comment (MartinSchuhmacher) [inline src/components/administration/AdminMigrationSection.unit.ts:583]: also using finalFinish here makes me more wanting the change, I suggested above ^^

**Review by MartinSchuhmacher — DISMISSED** [review]

**Pull request merged**